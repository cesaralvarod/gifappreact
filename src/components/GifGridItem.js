import React from "react";
import PropTypes from "prop-types";

// Styles

import "./GifGridItem.css";

function GifGridItem({ id, title, url }) {
  return (
    <li key={id} className="item">
      <img src={url} alt={title} />
      <p>{title}</p>
    </li>
  );
}

GifGridItem.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export default GifGridItem;
