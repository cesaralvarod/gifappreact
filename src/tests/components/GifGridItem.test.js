import React from "react";
import { shallow } from "enzyme";
import GifGridItem from "../../components/GifGridItem";

describe("Test in <GifGridItem />", () => {
  const id = "12312";

  const title = "A title";

  const url = "https://localhost.com";

  const wrapper = shallow(<GifGridItem id={id} title={title} url={url} />);

  test("Should show correctly the component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Should have one <p> tag with the title", () => {
    const p = wrapper.find("p");

    expect(p.text().trim()).toBe(title);
  });

  test("Should have one <img> tag with attributes src and alt", () => {
    const img = wrapper.find("img");

    expect(img.props().src).toBe(url);
    expect(img.props().alt).toBe(title);
  });

  test("Should be className 'item'", () => {
    expect(wrapper.props().className).toBe("item");
  });
});
