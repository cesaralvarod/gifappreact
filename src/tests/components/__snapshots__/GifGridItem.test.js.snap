// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifGridItem /> Should show correctly the component 1`] = `
<li
  className="item"
  key="12312"
>
  <img
    alt="A title"
    src="https://localhost.com"
  />
  <p>
    A title
  </p>
</li>
`;
